### PR TITLE
feat: add initial EIP-7805

### PR DESCRIPTION
**Pull request by jacobkaufmann**
### What was wrong?

N/A

### How was it fixed?

add initial implementation of EIP-7805.

#### Cute Animal Picture

`TODO`

### CODE REVIEW

File: src/ethereum/osaka/fork.py [hunk 1]
```diff
@@ -0,0 +1,1003 @@
...
+        # If the transaction is already present in the block, then skip.
+        if tx in transactions:
+            continue
+
+        block_env = deepcopy(block_env)
+        block_output = deepcopy(block_output)
+        try:
+            tx = decode_transaction(tx)
+            index = Uint(len(transactions))
+            process_transaction(block_env, block_output, tx, index)
```
Comment (jihoonsong) [inline src/ethereum/osaka/fork.py:923]: We don't execute IL transactions. We just need to check its nonce and balance, and see if there is sufficient gas left to include any missing IL transactions.

Current version of EIP is misleading and we have a pending PR that fixes it: https://github.com/ethereum/EIPs/pull/9381/files

Comment (jacobkaufmann) [reply src/ethereum/osaka/fork.py:923]: ah I see. I forgot about this PR, and so I was just going off what I saw in the EIP.

Comment (gurukamath) [reply src/ethereum/osaka/fork.py:923]: I would say in addition to checking for nonce and balance, would we need to perform some basic checks to make sure that the transaction is well formed? Including perhaps the checks that are performed in the [validate_transaction](https://github.com/ethereum/execution-specs/blob/68f3a1f1ccebc048daf5dfec58bf0325e5fcc1b8/src/ethereum/osaka/transactions.py#L191) function?

Comment (jihoonsong) [reply src/ethereum/osaka/fork.py:923]: The EL builds an IL with transactions from the mempool and each transaction is validated when added to the mempool. That being said, it makes sense to specify that IL transactions are intrinsically valid.

File: src/ethereum/osaka/fork.py [hunk 2]
```diff
@@ -705,6 +709,13 @@ def apply_body(
     for i, tx in enumerate(map(decode_transaction, transactions)):
         process_transaction(block_env, block_output, tx, Uint(i))
 
+    validate_inclusion_list(
```
Comment (jihoonsong) [inline src/ethereum/osaka/fork.py:712]: I want to share an idea about an edge case testing. It would be nice if we could test such situations that some withdrawal address become available to cover some IL transaction that they couldn't. That missing IL transaction should not reorg the block.

**Comment by fselmo** [comment]
closing in favor of #1928. Happy to re-open if this was in error.
